netbox/utilities/api.py

```diff
@@ -101,7 +102,7 @@ def to_internal_value(self, data):
                 except ValueError:
                     pass
 
-        if data not in self._choices:
+        if not isinstance(data, Hashable) or data not in self._choices:
```

Comment:
We can avoid the need for `Hashable` by just catching the `TypeError` as is done [in DRF native fields](https://github.com/encode/django-rest-framework/blob/master/rest_framework/fields.py#L686). Or better yet, just add `elif type(data) is not int` after the validation error for `dict`.